### ISSUE TITLE
Crash when trying to report unsupported type usage message for arg with default value

### ISSUE THREAD

**Issue by PiDelport** [swansonk14/typed-argument-parser#27]
Reproducing example:

```python
from pathlib import Path
from tap import Tap

class CrashingArgumentParser(Tap):
    some_path: Path = "some_path"

CrashingArgumentParser().parse_args()
```

Traceback:

```python-traceback
Traceback (most recent call last):
  File "tap_crash.py", line 10, in <module>
    CrashingArgumentParser().parse_args()
  File "…/site-packages/tap/tap.py", line 87, in __init__
    self._add_arguments()  # Adds all arguments in order to self
  File "…/site-packages/tap/tap.py", line 239, in _add_arguments
    self._add_argument(f'--{variable}')
  File "…/site-packages/tap/tap.py", line 183, in _add_argument
    arg_params = "required=True" if kwargs["required"] else f"default={getattr(self, variable)}"
KeyError: 'required'
```

The fix might be as simple as replacing `kwargs["required"]` with `kwargs.get("required", False)`?

**Comment by martinjm97** [comment]
Hi!

Thank you for reporting this to us and providing a succinct, reproducible example. I agree that the overarching problem here is that Path is not one of our supported types (https://github.com/swansonk14/typed-argument-parser#types) and that this error message is bad. I tested your suggested fix `kwargs.get("required", False)` and it looks like it does the trick!! 

We've pushed the fix and added a regression test (https://github.com/swansonk14/typed-argument-parser/commit/84bfc3722a59ee03b0b920e3f23c98194a546347) linking to your extremely clear and helpful issue.

Thanks again, 
Jesse
